### ISSUE TITLE
code3 Reference for functions

### ISSUE THREAD

**Issue by Ruthwik333** [HarshitKmr10/somethingsomething#3]
import pandas as pd
import numpy as np
import re
import unicodedata
import string
import pickle
from sklearn.preprocessing import LabelEncoder
from transformers import pipeline
import gc

# Load sentiment analysis model
pipe = pipeline(model="siebert/sentiment-roberta-large-english")

def encoding(text):
    text = unicodedata.normalize('NFKD', text)
    return text

def remove_non_ascii(text):
    return re.sub(r'[^\x00-\x7f]', r'', text)

def remove_punct(text):
    return text.translate(str.maketrans('', '', string.punctuation))

def truncate_text(text, max_length):
    # Truncate text to fit within max_length
    if len(text) <= max_length:
        return text
    else:
        return text[:max_length-3] + '...'

def predict_sentiment(text, results):
    # Clean the text
    cleaned_text = remove_punct(remove_non_ascii(encoding(text))).lower()
    # Truncate text to fit within the model's maximum sequence length
    cleaned_text = truncate_text(cleaned_text, 512)
    # Predict sentiment
    sentiment = pipe(cleaned_text)
    label = sentiment[0]['label']
    scores = sentiment[0]['score']  # Fix here, change 'results' to 'sentiment'
    return label, scores

def process_row(row, results):
    label, scores = predict_sentiment(row['Meeting Notes'], results)
    row['sentiment_label'] = label
    row['sentiment_scores'] = scores
    return row

# Read the Excel file into a DataFrame
df = pd.read_excel("FinaldataCRNEW.xlsx")

print(df.shape)
print(df.head())

# Drop rows with missing 'Meeting Notes' values
df.dropna(subset=['Meeting Notes'], inplace=True)
df.reset_index(inplace=True, drop=True)

print(df.shape)

# Process one row at a time
results = []
for index, row in df.iterrows():
    processed_row = process_row(row, results)
    results.append(processed_row)
    # Clear memory
    del processed_row
    gc.collect()

# Create a new DataFrame from the processed rows
df_processed = pd.DataFrame(results)

print(df_processed)
df_processed.to_csv('Totalnew.csv', index=False)
